test/lib/bamboo/email_test.exs

```diff
@@ -73,10 +23,176 @@ defmodule Bamboo.EmailTest do
     assert email.headers["Reply-To"] == "reply@foo.com"
   end
 
-  test "put_private/3 puts a key and value in the private attribute" do
-    email = new_email() |> put_private("foo", "bar")
+  describe "new_email/1" do
+    test "returns an Email struct" do
+      assert new_email() == %Bamboo.Email{
+               from: nil,
+               to: nil,
+               cc: nil,
+               bcc: nil,
+               subject: nil,
+               html_body: nil,
+               text_body: nil,
+               headers: %{}
+             }
+    end
+
+    test "can override email attributes with a map" do
+      email_attrs = %{to: "foo@bar.com", subject: "Cool Email"}
+      email = new_email(email_attrs)
+
+      assert email.to == "foo@bar.com"
```

Comment:
I don't think it's a big deal here, but just FYI exunit prints pretty helpful messages in the event of pattern match fails. So for instance you could do something like this...

```elixir
assert %Email{to: "foo@bar.com", subject: "Cool Email"} = email
```

...and you'd still get pretty handy failures if it didn't match

Reply:
Okay apparently I was wrong about this. My bad 😅